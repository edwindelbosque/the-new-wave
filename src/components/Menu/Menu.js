import React from 'react';
import './Menu.scss'

const Menu = ({ selectNews, categories }) => {

  const menuTabs = Object.keys(categories).splice(0, 5).map((key, index) => {
    return <li onClick={() => selectNews(categories[key])} key={index}>{key}</li>
  });

  return (
    <nav>
      <div className="fixed-nav">
        <h1>The New <span>wave </span></h1>
        <div className="menu-buttons-section">
          <ul>
            {menuTabs}
          </ul>
        </div>
      </div>
    </nav>
  )
}

export default Menu;