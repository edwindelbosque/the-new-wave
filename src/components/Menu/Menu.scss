nav {
  background-color: rgb(31, 31, 31);
  width: 300px;
  display: flex;
  flex-direction: column;
  align-items: center;

  h1 {
    margin: 20px auto;
    width: 182px;
    font-size: 30px;
    border: 2px solid rgb(255, 203, 32);
    color: rgb(255, 203, 32);
    border-radius: 1px;

    span {
      background-color: rgb(255, 203, 32);
      color: rgb(31, 31, 31);
      padding: 0 10px;
      padding: 0 57px;
    }
  }

  ul {
    display: flex;
    flex-direction: column; 
    align-items: flex-start;
    width: 265px;
    margin-top: 40px;
    height: 250px;
    padding: 0 35px;
    justify-content: space-around;
  }

  li {
    display: inline;
    color: rgb(245, 245, 245);
    padding: 5px 5px 5px 10px;
    font-weight: 600;
    font-size: 20px;
    transition: .15s;
    cursor: pointer;

    &:hover {
      padding-right: 35px;
      background-color: rgb(65, 30, 190);
      transition: .15s;
      border-right: 2px solid rgb(255, 203, 32);
    }
  }

}