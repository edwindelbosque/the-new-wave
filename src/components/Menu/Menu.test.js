import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('NewsArticle', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(
      <Menu
        selectNews="method"
        categories="array of categories" key="1"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});