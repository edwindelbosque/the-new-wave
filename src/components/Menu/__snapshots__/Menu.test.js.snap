// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data passed in correctly 1`] = `
<nav>
  <div
    className="fixed-nav"
  >
    <h1>
      The New 
      <span>
        wave 
      </span>
    </h1>
    <div
      className="menu-buttons-section"
    >
      <ul>
        <li
          key="0"
          onClick={[Function]}
        >
          0
        </li>
        <li
          key="1"
          onClick={[Function]}
        >
          1
        </li>
        <li
          key="2"
          onClick={[Function]}
        >
          2
        </li>
        <li
          key="3"
          onClick={[Function]}
        >
          3
        </li>
        <li
          key="4"
          onClick={[Function]}
        >
          4
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
